dm/dm/master/openapi_controller.go

```diff
@@ -275,10 +632,48 @@ func (s *Server) startTask(ctx context.Context, taskName string, sourceNameList
 		}
 		release()
 	}
-	return s.scheduler.UpdateExpectSubTaskStage(pb.Stage_Running, taskName, sourceNameList...)
+	return s.scheduler.UpdateExpectSubTaskStage(pb.Stage_Running, taskName, *req.SourceNameList...)
 }
 
-// nolint:unparam,unused
-func (s *Server) stopTask(ctx context.Context, taskName string, sourceNameList []string) error {
-	return s.scheduler.UpdateExpectSubTaskStage(pb.Stage_Stopped, taskName, sourceNameList...)
+// nolint:unparam
+func (s *Server) stopTask(ctx context.Context, taskName string, req openapi.StopTaskRequest) error {
+	// all subtasks for this task
+	if req.SourceNameList == nil || len(*req.SourceNameList) == 0 {
+		sourceNameList := openapi.SourceNameList(s.getTaskSourceNameList(taskName))
+		req.SourceNameList = &sourceNameList
+	}
+	// TODO(ehco): support stop req after https://github.com/pingcap/tiflow/pull/4601 merged
+	return s.scheduler.UpdateExpectSubTaskStage(pb.Stage_Stopped, taskName, *req.SourceNameList...)
+}
+
+// nolint:unparam
+func (s *Server) convertTaskConfig(ctx context.Context, req openapi.ConverterTaskRequest) (*openapi.Task, *config.TaskConfig, error) {
```

Comment:
where do we need this api?

Reply:
when user edit a task through taskConfig(v1 format config file) in WebUI